src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,24 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer extends Machine implements MachineProducer {
```

Comment:
The producer shouldn't extend Machine. If you think about inheritance, don't forget about Is-A, and Has-A rules. Is TruckProducer a Machine? Probably not. This is just an implementation of MachineProducer interface. Fix it all producers